src/Uno.UI.RemoteControl/HotReload/HotReloadStatusView.Entries.cs

```diff
@@ -0,0 +1,209 @@
+			_ => null
+		};
+		Description = Join("file", srvOp.FilePaths.Select(Path.GetFileName).ToArray()!);
+		Duration = srvOp.EndTime is not null ? srvOp.EndTime - srvOp.StartTime : null;
+
+		RaiseChanged();
+	}
+}
+
+internal record ApplicationEntry : HotReloadLogEntry
+{
+	public ApplicationEntry(HotReloadClientOperation localOp)
+		: base(EntrySource.Application, localOp.Id, localOp.StartTime)
+	{
+		Update(localOp);
+	}
+
+	internal void Update(HotReloadClientOperation localOp)
+	{
+		(IsSuccess, Icon) = localOp.Result switch
+		{
+			null => (default(bool?), EntryIcon.HotReload | EntryIcon.Loading),
+			HotReloadClientResult.Success => (true, EntryIcon.HotReload | EntryIcon.Success),
+			_ => (false, EntryIcon.HotReload | EntryIcon.Error)
+		};
+		Title = localOp.Result switch
+		{
+			null => "Processing...",
+			HotReloadClientResult.Success => "Update successful",
+			HotReloadClientResult.Failed => "An error occured",
+			_ => null
+		};
+		Description = Join("type", localOp.CuratedTypes, localOp.Types.Length);
+		Duration = localOp.EndTime is not null ? localOp.EndTime - localOp.StartTime : null;
+
+		RaiseChanged();
+	}
+}
+
+public enum EntrySource
+{
+	DevServer,
+	Engine,
+	Server,
+	Application
+}
+
+[Flags]
+public enum EntryIcon
+{
+	// Kind
+	Loading = 0x1,
+	Success = 0x2,
+	Warning = 0x3,
+	Error = 0x4,
+
+	// Source
+	Connection = 0x1 << 8,
+	HotReload = 0x2 << 8,
+}
+
+
+[Microsoft.UI.Xaml.Data.Bindable]
+internal record HotReloadLogEntry(EntrySource Source, long Id, DateTimeOffset Timestamp) : INotifyPropertyChanged
+{
+	/// <inheritdoc />
+	public event PropertyChangedEventHandler? PropertyChanged;
+
+	public bool? IsSuccess { get; set; }
+	public TimeSpan? Duration { get; set; }
+	public EntryIcon Icon { get; set; }
+
+	public string? Title { get; set; }
+	public string? Description { get; set; }
+	public string? ToastDescription => Description ?? Title;
+
+	public string TimeInfo => Duration switch
+	{
+		null => $"{Timestamp:T}",
+		{ TotalMilliseconds: < 1000 } ms => $"{ms.TotalMilliseconds:F0} ms - {Timestamp:T}",
+		{ } s => $"{s.TotalSeconds:N0} s - {Timestamp:T}",
+	};
+
+	protected void RaiseChanged()
+		=> PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(""));
+
+	protected static string? Join(string kind, string[] items, int? total = null, int max = 5)
+	{
+		const int maxLength = 70;
+
+		if (items is { Length: 0 } && total is null)
+		{
+			return null;
+		}
+
+		var sb = new StringBuilder(maxLength + 12 /* and xx more*/);
+		int count;
+		for (count = 0; count < Math.Min(items.Length, max); count++)
+		{
+			var item = items[count];
+			if (sb.Length + 2 /*, */ + item.Length < maxLength)
+			{
+				if (count is not 0) sb.Append(", ");
+				sb.Append(item);
+			}
+			else
+			{
+				break;
+			}
+		}
+
+		var remaining = total - count;
+		if (remaining > 0)
+		{
+			sb.Append((count, remaining) switch
+			{
+				(0, 1) => $"1 {kind}",
+				(0, _) => $"{remaining} {kind}s",
```

Comment:
No all words can be pluralized that way ;-)

Reply:
No, but the 2 that we are using are ;)